### PR TITLE
Adding SignedMessage

### PR DESCRIPTION
**Pull request by madhavajay**
## Description
- Added SignedMessage proto and class
- Added Tests for SignedMessage
- Added root_key to Duet Server
- Added ability to start Duet with id to Location mode
- Added from_string to UID for user friendly hex strings with test
- Fixed a few mypy type issues

## How has this been tested?
CI

## Checklist
- [x] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [x] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [x] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [x] My changes are covered by tests

### CODE REVIEW

**Comment by madhavajay** [comment]
Sorry, I tried to rebase and I think I made the commit graph... confusing.

**Comment by madhavajay** [comment]
This is rebased off https://github.com/OpenMined/PySyft/pull/3983 which is in turn rebased off syft_0.3.0.
You can probably just look at the last 2 commits since the previous ones are part of the fix-mypy and syft_0.3.0 branch already.

File: setup.cfg [hunk 1]
```diff
@@ -93,7 +93,7 @@ extras = True
 # in order to write a coverage file that can be read by Jenkins.
 # TODO (gmuraru) change cov-fail-under as soon as possible
 addopts =
-    --cov syft --cov-report term-missing --cov-fail-under 82 --no-cov-on-fail
```
Comment (iamtrask) [inline setup.cfg:96]: womp womp

Comment (madhavajay) [reply setup.cfg:96]: Sorry, I didn't see that revert in the rebase... 😬

**Pull request merged**